### PR TITLE
add current branch name on status bar

### PR DESCRIPTION
**Pull request by BoscoCHW**
Fixes https://github.com/jupyterlab/jupyterlab-git/issues/1100

@fcollonval 
Added current branch name on the status bar
I managed to style it but I'm not sure how to center the text vertically.
I should make the parent element a flex box and do align-item: center but I don't know how to select the parent class.
Also I'm not sure how to change the padding of the git icon so that the text could be closer to the icon.

Another issue is when navigate to another repo (change file directory), the branch name stays unless you click on the git extension icon on the left menu bar, which is a bit weird. 

---

Edited to link the related issue.

### CODE REVIEW

File: src/components/StatusWidget.tsx [hunk 1]
```diff
@@ -46,32 +47,39 @@ export class StatusWidget extends ReactWidget {
...
+                    ? `Git: ${this._trans.__('credentials required')}`
+                    : `Git: ${this._trans.__(this._status)}`
+                }
+              />
+            </Badge>
+            {this._model.currentBranch && (
+              <span className={currentBranchClass}>
+                {this._model.currentBranch.name}*
+              </span>
+            )}
```
Comment (fcollonval) [inline src/components/StatusWidget.tsx:81]: You should place this outside the `UseSignal` component as it does not need to be refreshed if the credentials status of the extension changes.

But you should wrap it in another `UseSignal` component to listen to `this._model.headChanged` to react when the git head is changed.

**Review by fcollonval — CHANGES_REQUESTED** [review]
Thanks @BoscoCHW 

> Added current branch name on the status bar I managed to style it but I'm not sure how to center the text vertically. I should make the parent element a flex box and do align-item: center but I don't know how to select the parent class. Also I'm not sure how to change the padding of the git icon so that the text could be closer to the icon.

You can add a class in `StatusWidget` (the parent element) within the [constructor](https://github.com/jupyterlab/jupyterlab-git/blob/master/src/components/StatusWidget.tsx#L27) by adding the following line:

```ts
this.addClass('jp-git-StatusWidget')
```

Note: by design it is not possible to select parent of an element with CSS selectors.

> Another issue is when navigate to another repo (change file directory), the branch name stays unless you click on the git extension icon on the left menu bar, which is a bit weird.

The default behavior is to not refresh the Git model status when the sidebar is hidden to avoid useless calls to the backend. You could solve this by forcing refreshing the branch calling [GitExtension.refreshBranch](https://github.com/jupyterlab/jupyterlab-git/blob/master/src/model.ts#L981) when we set a [new repository](https://github.com/jupyterlab/jupyterlab-git/blob/master/src/index.ts#L151).

**Comment by BoscoCHW** [comment]
@fcollonval 

I managed to style the status widget by adding a class like you suggested, but I was unable to reduce the right-padding of the git icon on the status bar because I'm afraid overriding the pre-set would lead to unexpected results.

File: src/components/StatusWidget.tsx [hunk 2]
```diff
@@ -41,39 +44,51 @@ export class StatusWidget extends ReactWidget {
...
+                  needsCredentials
+                    ? `Git: ${this._trans.__('credentials required')}`
+                    : `Git: ${this._trans.__(this._status)}`
+                }
+              />
+            </Badge>
+          )}
+        </UseSignal>
+
+        <UseSignal signal={this._model.headChanged}>
```
Comment (fcollonval) [inline src/components/StatusWidget.tsx:82]: :+1:

File: src/components/StatusWidget.tsx [hunk 3]
```diff
@@ -41,39 +44,51 @@ export class StatusWidget extends ReactWidget {
...
+          {() =>
+            this._model.currentBranch && (
+              <span className={currentBranchNameClass}>
+                {this._model.currentBranch.name}*
```
Comment (fcollonval) [inline src/components/StatusWidget.tsx:86]: Let's remove the star for now.

```suggestion
                {this._model.currentBranch.name}
```

**Review by fcollonval — CHANGES_REQUESTED** [review]
Let's remove the star next to the branch name. And it is good to go.

**Review by fcollonval — APPROVED** [review]
Thanks @BoscoCHW

**Pull request merged**